python/rikai/spark/functions/vision.py

```diff
@@ -425,3 +427,24 @@ def spectrogram_image(
         format=image_format,
         **image_kwargs,
     )
+
+
+@udf(returnType=ArrayType(ImageType(), True))
+def crop(img: Image, box: Union[Box2d, List[Box2d]]):
+    """Crop image specified by the bounding boxes, and returns the cropped
+    images.
+
+    Parameters
+    ----------
+    img : Image
+        An image object.
+    box : Box2d or List of Box2d
+        One bound-2d box or a list of such boxes
```

Comment:
Nit: maybe add that if a list of bounding boxes are given as input, the output will also be a list/array?

Reply:
It will return a list of images. Because udf can only support one type of output schema.